batched/dense/unit_test/Test_Utils.hpp

```diff
@@ -0,0 +1,105 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+#ifndef TEST_UTILS_HPP
+#define TEST_UTILS_HPP
+
+#include <Kokkos_Core.hpp>
+#include <iomanip>
+#include "KokkosBatched_Util.hpp"
+
+template <typename ExecutionSpace, typename AViewType, typename BViewType>
+bool allclose(const AViewType& a, const BViewType& b, double rtol = 1.e-5,
+              double atol = 1.e-8) {
+  constexpr std::size_t rank = AViewType::rank;
+  for (std::size_t i = 0; i < rank; i++) {
+    assert(a.extent(i) == b.extent(i));
+  }
+  const auto n = a.size();
+
+  auto* ptr_a = a.data();
+  auto* ptr_b = b.data();
+
+  int error = 0;
+  Kokkos::parallel_reduce(
+      Kokkos::RangePolicy<ExecutionSpace, Kokkos::IndexType<std::size_t>>{0, n},
+      KOKKOS_LAMBDA(const int& i, int& err) {
+        auto tmp_a = ptr_a[i];
```

Comment:
I think here you can just do `auto tmp_a = a(i)` no real need to use the pointer?

Reply:
For 1D Views, yes. The aim of this function is to apply on ND-views because we may need to compare between 2D Views or 3D Views. Thus, we need to access the internal data as 1D array.

On my way to remove this, I found that this is needed for the comparison of views stored in strided layout. As strided views are not deep-copy-able, we need an extra step of packing strided views into contiguous views before copying to host. This function allows the direct comparison on device so that we can avoid this packing

I am currently preparing the version without this, but tests have not finished yet. After checking the result, I will update this PR without `allclose` function.